packages/fx-core/src/core/middleware/consolidateLocalRemote.ts

```diff
@@ -307,8 +347,88 @@ async function consolidateLocalRemote(ctx: CoreHookContext): Promise<boolean> {
       );
     }
 
-    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateBackupConfig);
-    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateUpgrade);
+    if (needMigrateAadManifest) {
+      const permissionFilePath = path.join(inputs.projectPath as string, "permissions.json");
+
+      // add aad.template.file
+      const permissions = (await fs.readJson(permissionFilePath)) as Permission[];
+
+      const requiredResourceAccess = permissionsToRequiredResourceAccess(permissions);
+
+      const templatesFolder = getTemplatesFolder();
+      const aadManifestTemplatePath = `${templatesFolder}/${Constants.aadManifestTemplateFolder}/${Constants.aadManifestTemplateName}`;
+      const aadManifestJson: AADManifest = await fs.readJson(aadManifestTemplatePath);
+      aadManifestJson.requiredResourceAccess = requiredResourceAccess;
+      const aadManifestPath = path.join(
+        inputs.projectPath as string,
+        "templates",
+        "appPackage",
+        "aad.template.json"
+      );
+      const projectSettingsJson = await fs.readJson(projectSettingsPath);
+
+      if (projectSettingsJson.solutionSettings.capabilities.includes("Tab")) {
```

Comment:
Seems logic here are the same as in the AAD plugin, can we just call scaffold() here or maybe we can wrap it?

Reply:
Thanks, updated